allennlp/modules/seq2vec_encoders/wrapped_pytorch_rnn.py

```diff
@@ -0,0 +1,53 @@
+from typing import Type
+
+import torch
+
+from allennlp.common import Params
+from allennlp.common.checks import ConfigurationError
+from allennlp.modules import Seq2VecEncoder
+
+
+class WrappedPytorchRnn(Seq2VecEncoder):
+    """
+    Pytorch's RNNs have two outputs: the hidden state for every time step, and the hidden state at
+    the last time step for every layer.  We just want the second one as a single output.  This
+    wrapper pulls out that output, and allows us to instantiate the pytorch RNN from parameters.
+
+    This is actually slightly more complicated than that, because the LSTM's second output is a
+    tuple of (hidden state, cell state), and we just return the hidden state.
+
+    Also, there are lots of ways you could imagine going from an RNN hidden state at every
+    timestep to a single vector - you could take the last vector at all layers in the stack, do
+    some kind of pooling, take the last vector of the top layer in a stack, or many other  options.
+    We just take the final hidden state vector.  TODO(mattg): allow for other ways of wrapping
+    RNNs.
+    """
+
+    def __init__(self, module: torch.nn.modules.RNNBase) -> None:
+        super(WrappedPytorchRnn, self).__init__()
+        self._module = module
+
+    def get_output_dim(self) -> int:
+        return self._module.hidden_size * self._module.num_directions
+
+    def forward(self, inputs: torch.Tensor) -> torch.Tensor:  # pylint: disable=arguments-differ
+        return self._module(inputs)[0][-1]
+
+    class _Wrapper:
```

Comment:
This needs better documentation. It looks functional, but it also looks like we have a private class inside another class, which we call(?) to instantiate _itself_, where `from_params` is not a `classmethod` or a `staticmethod` as it is everywhere else but just calls itself to instantiate the outer class. There are also 2 `WrappedPytorchRnn` classes, which differ by 3 characters. 

How come this has to be inside `WrappedPytorchRnn`? Could it just be a separate private function which returns an instance of the `WrappedPytorchRnn` which is called in `from_params`?